### ISSUE TITLE
[a11y]:The 'headers' attribute value "expand" does not reference a valid 'id' in the table. 

### ISSUE THREAD

**Issue by preranagiri** [carbon-design-system/carbon#10920]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.52.0

### React version

17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

We use Data Table with expansion in our application and on running the IBM Equal Access Accessibility Checker we can see the following accessibility issue 
- the 'headers' attribute value "expand" does not reference a valid 'id' in the table.There is AVT1 issue raised against this.

![headerIssue](https://user-images.githubusercontent.com/13781018/156884985-8e552bf5-f719-4fd2-9f74-0c115033b9fd.png)

### CodeSandbox example

N/A

### Steps to reproduce

N/A

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by joshblack** [comment]
Hi there @preranagiri! 👋 I believe this was addressed over in: https://github.com/carbon-design-system/carbon/pull/10547 The storybook story should now be up-to-date for this use-case and teams should implement those changes on their end so that the AVT violation goes away 👍
